funnel/views/project.py

```diff
@@ -281,6 +299,42 @@ def rsvp(self):
     def rsvp_list(self):
         return {'project': self.obj, 'statuses': RSVP_STATUS}
 
+    @route('save', methods=['POST'])
+    @render_with(json=True)
+    @lastuser.requires_login
+    @requires_permission('view')
+    def save(self):
+        form = ProjectSaveForm()
+        if form.validate_on_submit():
+            proj_save = SavedProject.query.filter_by(user=current_auth.user, project=self.obj).first()
+            if form.save.data:
+                if proj_save is None:
+                    proj_save = SavedProject(user=current_auth.user, project=self.obj)
+                    form.populate_obj(proj_save)
+                    db.session.commit()
+                    return {'status': 'success'}
+                else:
+                    return {
+                        'status': 'error', 'error_identifier': 'project_save_exists',
+                        'error_description': _("You have already saved this project")
+                        }, 400
```

Comment:
This is not an error. For idempotency you return exactly like if the operation was successful.

Reply:
Use `'error'` as the key, not `'error_identifier'` (moot in this case because it's not to be used).